src/server/faculty/faculty.service.ts

```diff
@@ -36,4 +47,69 @@ export class FacultyService {
       .orderBy('"displayName"')
       .getRawMany();
   }
+
+  public async updateFacultyAbsences(
+    absenceReqInfo: Pick<Absence, 'id' | 'type'>
+  ): Promise<Absence> {
+    const existingAbsence = await this.absenceRepository
+      .findOne({
+        relations: ['faculty', 'semester'],
+        where: { id: absenceReqInfo.id },
+      });
+
+    const updateQuery = this.absenceRepository.createQueryBuilder()
+      .update(Absence);
+
+    // Are we going to have to do NO_LONGER_ACTIVE gymnastics?
+    if (
+      [existingAbsence.type, absenceReqInfo.type]
+        .some((value) => value === ABSENCE_TYPE.NO_LONGER_ACTIVE)
+    ) {
+      const { academicYear, term } = existingAbsence.semester;
+      // Looks like it - strap in!
+      const ids = this.absenceRepository.createQueryBuilder('a')
+        .select('a.id')
+        .leftJoin(Semester, 's', 'a."semesterId" = s.id')
+        .where('a."facultyId" = :facultyId')
+        .andWhere(new Brackets((q) => {
+          if (term === TERM.SPRING) {
+            q.where('s."academicYear" >= :academicYear');
+          } else if (term === TERM.FALL) {
+            q.where(
+              's."academicYear" > :academicYear'
+            ).orWhere(
+              's."academicYear" = :academicYear AND s."term" = :fall'
+            );
+          }
+        }));
+      updateQuery.where(`id IN (${ids.getQuery()})`);
+      // Changing TO NO_LONGER_ACTIVE
+      if (
+        existingAbsence.type !== ABSENCE_TYPE.NO_LONGER_ACTIVE
+        && absenceReqInfo.type === ABSENCE_TYPE.NO_LONGER_ACTIVE
+      ) {
+        updateQuery.set({ type: ABSENCE_TYPE.NO_LONGER_ACTIVE });
+      }
+
+      // Changing FROM NO_LONGER_ACTIVE
+      if (
+        existingAbsence.type === ABSENCE_TYPE.NO_LONGER_ACTIVE
+      && absenceReqInfo.type !== ABSENCE_TYPE.NO_LONGER_ACTIVE
+      ) {
+        updateQuery.set({ type: ABSENCE_TYPE.PRESENT });
+      }
```

Comment:
I think this case is more complicated, since it is possible to change from `NO_LONGER_ACTIVE` and a different absence type other than `PRESENT`. So if we change someone from `NO_LONGER_ACTIVE` to `PARENTAL_LEAVE` in Fall 2022, it should set Fall 2022 to `PARENTAL_LEAVE` and all future semesters to `PRESENT` --  right now it's setting all semesters to `PRESENT`. 

I think this would require two separate queries, one to update the requested semester, and another to update all future semester.

Reply:
Oh, I see. My understanding from the ticket was that the current semester would be set to `ABSENCE_TYPE.PRESENT` as well as all future semesters. If that's not the case then it will indeed require two queries, yes.

Reply:
Alright, I think I've updated this in 4a2ddc0b34a1c94f1caf9fc2f57c03801b7a7281